CondFormats/SiPixelObjects/interface/SiPixelFEDChannelContainer.h

```diff
@@ -0,0 +1,50 @@
+#ifndef CondFormats_SiPixelObjects_SiPixelFEDChannelContainer_h 
+#define CondFormats_SiPixelObjects_SiPixelFEDChannelContainer_h
+
+#include "CondFormats/Serialization/interface/Serializable.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/SiPixelDetId/interface/PixelFEDChannel.h"   // N.B. a DataFormat is serialized here (need for dedicated serialization rules, see CondFormats/External/interface/PixelFEDChannel.h)
+
+#include <map>
+#include <string>
+#include <vector>
+
+class SiPixelFEDChannelContainer{
+public:
+  typedef std::map<DetId,std::vector<PixelFEDChannel> > SiPixelFEDChannelCollection;
```

Comment:
This choice requires the persistency of a non-CondFormats class ( correctly treated above in the CondFormats/External package ). Since the DataFormats class layouts are not guaranteed to be 'frozen', this brings potentially some non-zero ( although presumably very low ) risk of schema breaking in the future.  
For this reason, the requirement of this dependency is expected to be well justified.

Reply:
Hi @ggovi, 
thanks for your comment. 
As I highlighted in the PR description: we don't foresee to ever have to change the content of the `DataFormat`, so schema evolution is implicitly avoided.
The current format of the `edmnew::DetSetVector<PixelFEDChannel>` contains "redundant" information to find a FED channel (`DetId`,`FEDId`, `LinkId`, `first_roc` and `last_roc`). This is fine to describe any FED channel in phase-0 and phase-I detectors.
The main con to code a dedicated container class within `CondFormat` (that has the advantage that is guaranteed not to break the schema), is that we'll need to code within each piece of client code (that understands the  `PixelFEDChannel` object) a conversion between the `CondFormat` and the `DataFormat`.